frontend/tests/components/session-timeout.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { createRoutesStub } from 'react-router';
+
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import { useIdleTimer } from 'react-idle-timer';
+import { describe, expect, it, vi } from 'vitest';
+
+import type { SessionTimeoutProps } from '~/components/session-timeout';
+import SessionTimeout from '~/components/session-timeout';
+
+describe('SessionTimeout', () => {
+  const mockActivate = vi.fn();
+  const mockIsPrompted = vi.fn();
+  const mockGetRemainingTime = vi.fn();
+
+  vi.mock('react-idle-timer');
+
+  vi.mocked(useIdleTimer, { partial: true }).mockReturnValue({
+    activate: mockActivate,
+    isPrompted: mockIsPrompted,
+    getRemainingTime: mockGetRemainingTime,
+  });
+
+  const setup = (props: Partial<SessionTimeoutProps> = {}) => {
+    const defaultProps = {
+      promptBeforeIdle: 30000,
+      timeout: 60000,
+      onSessionEnd: vi.fn(),
+      onSessionExtend: vi.fn(),
+    };
+
+    const RoutesStub = createRoutesStub([
+      {
+        Component: () => <SessionTimeout {...defaultProps} {...props} />,
+        path: '/',
+      },
+    ]);
+
+    return render(<RoutesStub />);
+  };
+
+  it('should render the session timeout dialog when prompted', () => {
+    mockIsPrompted.mockReturnValue(true);
+
+    setup();
+
+    expect(screen.getByText('session-timeout.header')).toBeInTheDocument();
+    expect(screen.getByText('session-timeout.end-session')).toBeInTheDocument();
+    expect(screen.getByText('session-timeout.continue-session')).toBeInTheDocument();
+  });
+
+  it('should call `onSessionEnd` when the "End Session" button is clicked', () => {
+    const onSessionEnd = vi.fn();
+    mockIsPrompted.mockReturnValue(true);
+
+    setup({ onSessionEnd });
+
+    const endSessionButton = screen.getByText('session-timeout.end-session');
+
+    act(() => {
+      fireEvent.click(endSessionButton);
+    });
+
+    expect(onSessionEnd).toHaveBeenCalled();
+  });
+
+  it('should call `onSessionExtend` when the "Continue Session" button is clicked', () => {
+    const onSessionExtend = vi.fn();
+    mockIsPrompted.mockReturnValue(true);
+
+    setup({ onSessionExtend });
+
+    const continueSessionButton = screen.getByText('session-timeout.continue-session');
+
+    act(() => {
+      fireEvent.click(continueSessionButton);
+    });
+
+    expect(onSessionExtend).toHaveBeenCalled();
+  });
+
+  it('should activate the IdleTimer on location change', () => {
+    mockIsPrompted.mockReturnValue(false);
+
+    setup();
+
+    expect(mockActivate).toHaveBeenCalled();
+  });
+
+  it('should update the remaining time every second', () => {
+    mockIsPrompted.mockReturnValue(true);
+    mockGetRemainingTime.mockReturnValue(50000);
+
+    vi.useFakeTimers({ shouldAdvanceTime: true });
+
+    setup();
+
+    expect(screen.getByText(`session-timeout.description{"timeRemaining":"0:50"}`)).toBeInTheDocument();
```

Comment:
This is not going to work with the current implementation of the react-i18next automock. I will update the mock to append the placeholder values to each string and you can cherry pick that once merged.

Reply:
Cherry pick 8323b31 into your branch and you will be able to fix this test.

Reply:
Once you cherry pick that commit, you can use this:

``` typescript
    expect(screen.queryByText('{"key":"gcweb:session-timeout.description","options":{"timeRemaining":"0:50"}}')).not.toBeNull();

```